### ISSUE TITLE
Using bazel to cross-compile tensorflow for other targets.

### ISSUE THREAD

**Issue by gopinath-r** [bazelbuild/bazel#3692]
I am trying to cross compile tensorflow to create a pip package by following the bellow threads.

https://github.com/bazelbuild/bazel/wiki/Building-with-a-custom-toolchain#writing-the-crosstool
https://github.com/bazelbuild/bazel/wiki/About-the-CROSSTOOL

By following the steps mentioned in the above pages, I got some errors and I could resolve some errors by going through the following thread 

https://github.com/bazelbuild/bazel/issues/1353

The command I am using to build pip package is 
bazel build --crosstool_top=//tools/arm_compiler:toolchain --cpu=armeabi-v7a //tensorflow/tools/pip_package:build_pip_package

Now i am getting this error 
ERROR: /home/gopinathr/Documents/work/tensorflow/tools/arm_compiler/BUILD:51:1: in cc_toolchain rule //tools/arm_compiler:cc-compiler-armeabi-v7a: The package '@linaro_linux_gcc_repo' is not valid

ERROR: Analysis of target '//tensorflow/tools/pip_package:build_pip_package' failed; build aborted

The Build,Crosstool,Workspace files i am using are,
[BUILD.txt](https://github.com/bazelbuild/bazel/files/1279947/BUILD.txt)
[CROSSTOOL.txt](https://github.com/bazelbuild/bazel/files/1279948/CROSSTOOL.txt)
[WORKSPACE.txt](https://github.com/bazelbuild/bazel/files/1279950/WORKSPACE.txt)

Could anyone help me out what is wrong.

**Comment by philwo** [comment]
@lberki Any idea?

**Comment by lberki** [comment]
in my professional opinion, I don't know. What does `blaze query @linaro_linux_gcc_repo//...` say? In addition to the "package is not valid" error message, there should be something else that tells you what's wrong.

**Comment by gopinath-r** [comment]
@lberki thanks for the reply... The error was because of the irregular names used in the configuration..
It resolved now.

**Comment by tollytao** [comment]
@gopinath-r  I also try to do cross-compile for arm 32 bit, I have some questions: 
1   CROSSTOOL and BUILD will located in  tensorflow/tools/arm_compiler/, right ?  
2   WORKSPACE located in tensorflow/？ 
3   Where  can I got  arm_compiles , I copy it from bazel/testdata/bazel_toolchain_test_data/tools/arm_compiler， but it showed some errors . 
Waiting for your reply!

**Comment by gopinath-r** [comment]
@tollytao Are you trying to cross compile tensorflow or some other library

**Comment by gopinath-r** [comment]
@tollytao 
I couldn't succeed in cross-compilation of tensorflow for arm 32 bit. So I compiled the tensorflow from source on the board.
I followed raspberry pi steps and some modification because my cpu is cortex-a15 which is different from raspberry pi and i could get the tensorflow pip package successfully.

I cross-compiled tensorflow-lite which doesn't have python bindings successfully but i was using makefile to do that.

I am not much familiar with bazel cross compilation so i stopped working on cross compilation of tensorflow pip package.

But still if you want to try, I can tell you what are the things i did for cross compilation using bazel.
